forge-game/src/main/java/forge/game/card/Card.java

```diff
@@ -1577,6 +1578,11 @@ public final void subtractCounter(final CounterType counterName, final int n) {
         }
         runParams.put(AbilityKey.CounterAmount, delta);
         getGame().getTriggerHandler().runTrigger(TriggerType.CounterRemovedOnce, AbilityKey.newMap(runParams), false);
+
+        if (counterName.is(CounterEnumType.DEFENSE) && newValue == 0) {
```

Comment:
This is certainly not 100% correct, since I'm guessing only Battles having their Defense counters reduced to 0 trigger Defeated.